### PR TITLE
Improve building times of IPU docker image

### PR DESCRIPTION
**Pull request by otaj**
## What does this PR do?

IPU docker image build quite often fails because it is too slow. This PR rearranges couple things to improve build times, whereas the image should remain mostly the same

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: dockers/base-ipu/Dockerfile [hunk 1]
```diff
@@ -56,13 +57,12 @@ ENV \
 
 COPY environment.yml environment.yml
 
-RUN conda create -y --name $CONDA_ENV python=${PYTHON_VERSION} pytorch=${PYTORCH_VERSION} cudatoolkit=${CUDA_VERSION} -c pytorch && \
-    conda init bash && \
+RUN conda init bash && \
```
Comment (justusschock) [inline dockers/base-ipu/Dockerfile:60]: one general question: why don't we start with a miniconda image in the first place?

**Review by justusschock — APPROVED** [review]

File: dockers/base-ipu/Dockerfile (hunk 1 above)
Comment (otaj) [reply dockers/base-ipu/Dockerfile:60]: No idea. I can refactor the image to it, but that's not work for this PR, in this one I tried to just improve building times while keeping the image as similar to the original as possible

Comment (justusschock) [reply dockers/base-ipu/Dockerfile:60]: agreed, but this would further reduce building time on our end :)

Comment (carmocca) [reply dockers/base-ipu/Dockerfile:60]: Resolving to unblock master. We can consider further reductions separately

**Comment by akihironitta** [comment]
Thank you @otaj! Will follow up in my PR #14057 to further reduce the build time.

**Pull request merged**